### ISSUE TITLE
Undoable mutations not executed if user closes browser tab too soon

### ISSUE THREAD

**Issue by zenflow** [marmelab/react-admin#8505]
**What you were expecting:**
I was expecting that undoable mutations are executed unless the user clicks "UNDO" on the "Element updated" notification.

**What happened instead:**
If user closes the browser tab before the "Element updated" notification disappears (that being the time when the mutation **would** be executed), the mutation is not executed. The user has no indication that the mutation didn't go through, and would assume that it did.

**Steps to reproduce:**
1. Make an edit to a resource that does undoable edits
2. Close the browser tab immediately after that (before the "Element updated" notification disappears)
3. Open another browser tab and view the resource that you just edited
4. Observe that the edit was never applied

**Related code:**

n/a

**Other information:**

none

**Environment**

* React-admin version: 4.5.2
* Last version that did not exhibit the issue (if applicable): N/A
* React version: 18.2.0
* Browser: Chrome (or any)

**Comment by zenflow** [comment]
I think a `beforeunload` listener could solve the problem by having the browser show this dialog if there are undoable mutations that haven't executed when user tries to close the browser tab: 
![image](https://user-images.githubusercontent.com/3198597/207702024-cc37b1a2-2b2c-411c-80b9-cf4be9d692e6.png)

**Comment by slax57** [comment]
You're right, we should warn the user that their changes have not been persisted yet. I'll label this issue as bug. Thanks.
